### PR TITLE
Homework 7 task 1

### PR DESCRIPTION
**Pull request by ivanzuev78**

### CODE REVIEW

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_7/test_task_1.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def test_find_occurrences_element_in_key():
+
+    tree = {
+        "elem": "elem",
+        "elem_in_dict": {"elem": "elem", 2: "bad_elem"},
+    }
+
+    assert find_occurrences(tree, "elem") == 2
```
[the remark this answers is not in the document]
Comment (ivanzuev78) [reply HomeWork_7/test_task_1.py:39]: I changed this test. Now we have two keys and no one of them was counted.

File: HomeWork_7/test_task_1.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        "elem_in_list": [elem_to_search, "bad_elem"],
+        "elem_in_dict": {1: elem_to_search, 2: "bad_elem"},
+        "elem_in_set": {elem_to_search, "bad_elem"},
+        "elem_in_tuple": (elem_to_search, "bad_elem"),
+    }
+
+    assert find_occurrences(tree, elem_to_search) == 5
+
+
+@pytest.mark.parametrize("elem_to_search", ({1, "elem"}, {1: 2, 2: True}))
```
[the remark this answers is not in the document]
Comment (ivanzuev78) [reply HomeWork_7/test_task_1.py:19]: Test added.

File: HomeWork_7/hw1.py [hunk 3]
```diff
@@ -29,7 +28,22 @@
 
 
 def find_occurrences(tree: dict, element: Any) -> int:
-    ...
+    counter = 0
+
+    def look_into_collection(collection_to_discover: Union[list, tuple, dict, set]):
+        nonlocal counter
```
[the remark this answers is not in the document]
Comment (ivanzuev78) [reply HomeWork_7/hw1.py:34]: Yes

File: HomeWork_7/hw1.py [hunk 4]
```diff
@@ -6,31 +6,20 @@
...
-        },
-    },
-    "fourth": "RED",
-}
+from typing import Any, Union
 
 
 def find_occurrences(tree: dict, element: Any) -> int:
-    ...
+    def look_into_collection(collection_to_discover: Union[list, tuple, dict, set]):
```
Comment (Cjkjvfnby) [inline HomeWork_7/hw1.py:13]: https://www.oxfordlearnersdictionaries.com/definition/english/discover

File: HomeWork_7/test_task_1.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        "elem_in_list": [elem_to_search, "bad_elem"],
+        "elem_in_dict": {1: elem_to_search, 2: "bad_elem"},
+        "elem_in_set": {elem_to_search, "bad_elem"},
+        "elem_in_tuple": (elem_to_search, "bad_elem"),
+    }
+
+    assert find_occurrences(tree, elem_to_search) == 5
+
+
+@pytest.mark.parametrize("elem_to_search", [{1, "elem"}, {1: 2, 2: True}])
```
Comment (Cjkjvfnby) [inline HomeWork_7/test_task_1.py:19]: What is the difference between hashable and unhashible elements in this task context?

File: HomeWork_7/test_task_1.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def test_find_occurrences_element_in_key():
+
+    tree = {
+        "elem": "first key",
+        "elem_in_dict": {"elem": "second key", 2: "bad_elem"},
+    }
+
+    assert find_occurrences(tree, "elem") == 0
```
Comment (Cjkjvfnby) [inline HomeWork_7/test_task_1.py:39]: `test_find_occurrences_element_in_key`  Do we found this elements?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_7/test_task_1.py (hunk 6 above)
Comment (ivanzuev78) [reply HomeWork_7/test_task_1.py:39]: No. As I remember, we shouldn't search elements in keys.

File: HomeWork_7/test_task_1.py (hunk 5 above)
Comment (ivanzuev78) [reply HomeWork_7/test_task_1.py:19]: `set` doesn't accept hashable elements. 
I changed tests.

File: HomeWork_7/hw1.py (hunk 4 above)
Comment (ivanzuev78) [reply HomeWork_7/hw1.py]: Thank you.
Changed.

File: HomeWork_7/test_task_1.py (hunk 6 above)
Comment (Cjkjvfnby) [reply HomeWork_7/test_task_1.py:37]: The test name should show me your intention.  `test_find_occurrences_element_in_key` means for me that we are searching in key.  If you write the test to show that element in key is ignored just name the test in that way.

Comment (Cjkjvfnby) [reply HomeWork_7/test_task_1.py:37]: @ivanzuev78 please fix it

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**